src/adapter/etl-adapter-avro/src/Flow/ETL/Adapter/Avro/FlixTech/AvroExtractor.php

```diff
@@ -41,7 +47,14 @@ public function extract(FlowContext $context) : \Generator
                     $row['_input_file_uri'] = $filePath->uri();
                 }
 
-                yield array_to_rows($row, $context->entryFactory());
+                $signal = yield array_to_rows($row, $context->entryFactory());
+                $this->countRow();
+
+                if ($signal === Signal::STOP || $this->reachedLimit()) {
+                    $context->streams()->close($this->path);
+
+                    return;
```

Comment:
in all file-based ones it could be replaced with:
```suggestion
                    break 2;
```
along with that closing the stream could be removed as would be done at the end of that method.

Reply:
I started from this approach but return is easier to understand and less problematic in case of additional logic after the loop. Return is more readable at least for me